tests/rptest/tests/e2e_shadow_indexing_test.py

```diff
@@ -88,6 +90,43 @@ def tearDown(self):
 
 
 class EndToEndShadowIndexingTest(EndToEndShadowIndexingBase):
+    @cluster(num_nodes=5)
+    @matrix(cloud_storage_type=get_cloud_storage_type())
+    def test_reset(self, cloud_storage_type):
+        brokers = self.redpanda.started_nodes()
+
+        self.start_producer()
+        produce_until_segments(
+            redpanda=self.redpanda,
+            topic=self.topic,
+            partition_idx=0,
+            count=10,
+        )
+        original_snapshot = self.redpanda.storage(
+            all_nodes=True).segments_by_node("kafka", self.topic, 0)
+        self.kafka_tools.alter_topic_config(
+            self.topic,
+            {
+                TopicSpec.PROPERTY_RETENTION_LOCAL_TARGET_BYTES:
+                5 * self.segment_size,
+            },
+        )
+
+        wait_for_removal_of_n_segments(redpanda=self.redpanda,
+                                       topic=self.topic,
+                                       partition_idx=0,
+                                       n=6,
+                                       original_snapshot=original_snapshot)
+        # Wait for there to be some segments.
+        def manifest_has_segments():
+            s3_snapshot = BucketView(self.redpanda, topics=self.topics)
+            manifest = s3_snapshot.manifest_for_ntp(self.s3_topic_name,
+                                                    0)
+            return ("segments" in manifest, manifest)
+
+        manifest = wait_until_result(manifest_has_segments, timeout_sec=30, backoff_sec=1)
+        self.redpanda._admin.unsafe_reset_cloud_metadata(self.s3_topic_name, 0, json.dumps(manifest))
```

Comment:
We should probably try to produce and consume after re-setting to ensure some basic functionality.